src/library/scala/collection/immutable/StrictOptimizedSeqOps.scala

```diff
@@ -41,9 +40,7 @@ trait StrictOptimizedSeqOps[+A, +CC[_], +C]
   override def updated[B >: A](index: Int, elem: B): CC[B] = {
     if (index < 0) throw new IndexOutOfBoundsException(s"$index is out of bounds (min 0, max ${if (knownSize>=0) knownSize else "unknown"})")
     val b = iterableFactory.newBuilder[B]
-    if (knownSize >= 0) {
-      b.sizeHint(size)
-    }
+    b.sizeHint(this)
```

Comment:
Assuming knownSize is size if known, not "I know I'm size 30, but maybe size says I'm size 40 because it is lazily updated."